src/configure/auto_mozilla.rs

```diff
@@ -38,59 +36,219 @@ enum MozConfigTag {
-    let mut reader = quick_xml::Reader::from_str(xml_raw);
-    reader.trim_text(true);
+/// Parses a single IncomingServer or OutgoingServer section.
+fn parse_server<B: BufRead>(
+    reader: &mut quick_xml::Reader<B>,
+    server_event: &BytesStart,
+) -> Result<Option<Server>, quick_xml::Error> {
+    let end_tag = String::from_utf8_lossy(server_event.name())
+        .trim()
+        .to_lowercase();
+
+    let typ = server_event
+        .attributes()
+        .find(|attr| {
+            attr.as_ref()
+                .map(|a| String::from_utf8_lossy(a.key).trim().to_lowercase() == "type")
+                .unwrap_or_default()
+        })
+        .map(|typ| {
+            typ.unwrap()
+                .unescape_and_decode_value(reader)
+                .unwrap_or_default()
+                .to_lowercase()
+        })
+        .unwrap_or_default();
+
+    let mut hostname = None;
+    let mut port = None;
+    let mut sockettype = Socket::Automatic;
+    let mut username = None;
 
+    let mut tag_config = MozConfigTag::Undefined;
+    let mut buf = Vec::new();
+    loop {
+        match reader.read_event(&mut buf)? {
+            Event::Start(ref event) => {
+                let tag = String::from_utf8_lossy(event.name()).trim().to_lowercase();
+
+                if tag == "hostname" {
+                    tag_config = MozConfigTag::Hostname;
+                } else if tag == "port" {
+                    tag_config = MozConfigTag::Port;
+                } else if tag == "sockettype" {
+                    tag_config = MozConfigTag::Sockettype;
+                } else if tag == "username" {
+                    tag_config = MozConfigTag::Username;
+                }
+            }
+            Event::End(ref event) => {
+                let tag = String::from_utf8_lossy(event.name()).trim().to_lowercase();
+
+                if tag == end_tag {
+                    break;
+                }
+            }
+            Event::Text(ref event) => {
+                let val = event
+                    .unescape_and_decode(reader)
+                    .unwrap_or_default()
+                    .trim()
+                    .to_owned();
+
+                match tag_config {
+                    MozConfigTag::Hostname => hostname = Some(val),
+                    MozConfigTag::Port => port = Some(val.parse().unwrap_or_default()),
+                    MozConfigTag::Username => username = Some(val),
+                    MozConfigTag::Sockettype => {
+                        let val_lower = val.to_lowercase();
+                        if val_lower == "ssl" {
+                            sockettype = Socket::SSL;
+                        }
+                        if val_lower == "starttls" {
+                            sockettype = Socket::STARTTLS;
+                        }
+                        if val_lower == "plain" {
+                            sockettype = Socket::Plain;
+                        }
```

Comment:
what about the case it is aynthing else?

Reply:
I set it to `Socket::Automatic` now. But this is old code, it was only moved around. Making parsing more strict or improving it in other ways is not the goal of this PR.